Add opacity to highway=proposed for less prominent rendering

sb12 opened the pull request:
Based on the discussion in and supersedes #863.
At least partly resolves #826 and #345.
This does not add special styles for tunnels or bridges and might need some rework, when the 'proposed=*' key is available in the database.

Rendering example:
![proposed](https://cloud.githubusercontent.com/assets/5783139/5691170/c6471d7e-98b9-11e4-8f7b-7fcf8e401d17.png)

mkoniecz commented:
@sb12 "The construction rules for small roads are strange(...)" comment should appear before construction rules, not before proposed rules.

sb12 commented:
@mkoniecz thanks for spotting it. It should be fixed now.

mkoniecz commented:
Is it intentional that highway=proposed is no longer rendered on z12? Current code is suggesting that it is present at this zoom level, but effects are different.

BTW, I am supporting dropping rendering from z12.

![highway proposed name eeeeee eeeeee eeeeee ref 1 ele 8000 way master - sb12_highway_proposed2](https://cloud.githubusercontent.com/assets/899988/5691990/85215472-98e1-11e4-818e-eb8ceffc7558.png)

mkoniecz commented:
From test on real data - it seems this transparency is really strong, it basically removes rendering except name.

![36 097753 139 6267966 highway proposed master - sb12_highway_proposed2](https://cloud.githubusercontent.com/assets/899988/5692010/14aba560-98e3-11e4-8731-05399b7fdda8.png)

![50 0746011 19 8176432 highway proposed master - sb12_highway_proposed2](https://cloud.githubusercontent.com/assets/899988/5691998/7f4277ec-98e2-11e4-903e-4aad30136a11.png)

daganzdaanda commented:
Thanks for the real world examples. I agree, if it's so very transparent, we can just as well leave it out. Maybe 0.3 opacity would already be enough? 
OTOH, I'd like to have the labels *less* visible. In z<=14 I wouldn't render a label for proposed roads at all. And after that, grey instead of black would probably fit the new style better.

pnorman commented on roads.mss:
```diff
@@ -699,6 +699,44 @@
+        line-width: 2;
+        line-color: white;
+        line-opacity: 0.2;
+        a/line-width: 2;
```
can you use more descriptive names than a b and c?

sb12 commented:
Thank you for your feedback.

@daganzdaanda 
 >Thanks for the real world examples. I agree, if it's so very transparent, we can just as well leave it out. Maybe 0.3 opacity would already be enough?
OTOH, I'd like to have the labels less visible. In z<=14 I wouldn't render a label for proposed roads at all. And after that, grey instead of black would probably fit the new style better.

I changed the opacity to 0.3 and added an opacity to the labels as well. I also removed labels for zoom levels <=14:
![proposedheidelberg14](https://cloud.githubusercontent.com/assets/5783139/5789043/b9f6305e-9e5a-11e4-8790-461a888727c6.png)
![proposedheidelberg16](https://cloud.githubusercontent.com/assets/5783139/5789044/b9f94ae6-9e5a-11e4-98ce-d452a9ed4498.png)

@mkoniecz maybe you could render some more examples with your tool. I tried to use it but I didn't know how to and couldn't find any documentation. (Does it work on Linux?)

> BTW, I am supporting dropping rendering highway=proposed from z12.

I agree with you, so I removed rendering in zoom 12 and also changed the rendering in zoom level 13 to the one which was used in zooom level 12 before. This improves rendering of planned residential highways a lot. (I'm planning to provide a separate pull request for roads under construction which have a similar problem.)
Example:
![proposedbruchsal](https://cloud.githubusercontent.com/assets/5783139/5789046/bf549f90-9e5a-11e4-9ec3-859208a82da0.png)

@pnorman I improved the code to be more descriptive.

mkoniecz commented:
> I tried to use it but I didn't know how to and couldn't find any documentation. (Does it work on Linux?)

@sb12 - I was unsure whatever anybody will want to use it, so at this moment there is no documentation. Also, some features that are necessary to make it usable for other people are missing at this moment (it should change in the near future). It works on Linux.

mkoniecz commented:
I am happy about results.

https://cloud.githubusercontent.com/assets/899988/5789450/6a14b27a-9e6b-11e4-8f79-743062e8d690.png

https://cloud.githubusercontent.com/assets/899988/5789448/6a0d2424-9e6b-11e4-82e8-9a5c5c8533ae.png

https://cloud.githubusercontent.com/assets/899988/5789451/6a16c25e-9e6b-11e4-9cc3-5a29b9ae7f6d.png

https://cloud.githubusercontent.com/assets/899988/5789449/6a1419e6-9e6b-11e4-935a-78d7a1e171be.png

https://cloud.githubusercontent.com/assets/899988/5789453/6a18f7a4-9e6b-11e4-817d-99aafb5e0687.png

https://cloud.githubusercontent.com/assets/899988/5789452/6a17c956-9e6b-11e4-8a73-96f6acca7528.png

math1985 commented:
Looks goood to me, and can be merged as far as I am concerned.

matkoniecz commented:
@sb12 - I finally released my helper script in form that has chance to be usable for other people (https://github.com/matkoniecz/CartoCSSHelper).
